packages/kolme-cli/src/chain_query.rs

```diff
@@ -0,0 +1,223 @@
+use anyhow::{anyhow, bail, ensure, Context, Result};
+use reqwest::{Client, Url};
+use serde::Deserialize;
+
+use crate::RequestBuilderExt;
+
+pub struct ChainApi {
+    client: Client,
+    url: Url,
+}
+
+#[derive(Deserialize)]
+#[allow(dead_code)]
+pub struct ChainVersion {
+    pub code_version: String,
+    pub chain_version: String,
+    pub next_height: BlockHeight,
+}
+
+#[allow(dead_code)]
+pub struct BlockResponse {
+    pub code_version: String,
+    pub chain_version: String,
+    pub block_height: BlockHeight,
+}
+
+pub struct ForkInfo {
+    pub first_block: BlockHeight,
+    pub last_block: BlockHeight,
+}
+
+#[derive(Deserialize, Copy, Clone, Debug)]
+pub struct BlockHeight(pub u32);
```

Comment:
I'm surprised you're defining your own type here instead of using the one in `kolme` itself, any motivation for that decision?